### ISSUE TITLE
Require unlocking to use USB in fastbootd/recovery when bootloader is locked

### ISSUE THREAD

**Issue by flawedworld** [GrapheneOS/os-issue-tracker#2142]

**Comment by secrecyflag** [comment]
Do you mean running `adb reboot bootloader` needs unlocking?

**Comment by r3g-5z** [comment]
No, that's the firmware fastboot. The issue is about userspace fastbootd/recovery which is used for things like sideloading updates.

**Comment by secrecyflag** [comment]
@r3g-5z so basically `adb reboot fastboot` needs unlocking?

**Comment by juksa010** [comment]
I'm also wondering why this is necessary when we already have a locked bootloader? Why the need for extra unlocking when we have already a locked bootloader to begin with?

**Comment by thestinger** [comment]
Seems overcomplicated when the only available functionality is in recovery for sideloading signed updates which is similar to the update client.

**Comment by thestinger** [comment]
fastbootd already checks for locking.
